scripts/common.sh

```diff
@@ -193,16 +193,56 @@ function msg_on_error
     fi
 }
 
+
 function diff_values
 {
     (( $# >= 2 )) || fail "diff_values requires at least 2 arguments."
 
     local value1="$1"
     local value2="$2"
-    shift
-    shift
+    shift 2
 
-    diff --unified=0 <(echo "$value1") <(echo "$value2") "$@"
+    if ! diff --unified=0 <(echo "$value1") <(echo "$value2") "$@"
+    then
+        if [ "${DIFFVIEW:-}" == "" ]
+        then
+            printError "ERROR: values differ:"
+            printError "Expected:"
+            printError "${value1}"
+            printError "Obtained:"
+            printError "${value2}"
+        else
+            # Use user supplied diff view binary
+            printError "ERROR: values differ."
+            "$DIFFVIEW" "${file1}" "${file2}"
+        fi
+        return 1
+    fi
+}
+
+function diff_files
+{
+    (( $# >= 2 )) || fail "diff_files requires at least 2 arguments."
+
+    local file1="$1"
+    local file2="$2"
+
+    if ! diff "${file1}" "${file2}"
+    then
+        if [ "${DIFFVIEW:-}" == "" ]
+        then
+            printError "ERROR: files differ: ${file1} vs. ${file2}"
+            printError "Expected:"
+            cat "${file1}" >&2
+            printError "Obtained:"
+            cat "${file2}" >&2
```

Comment:
Is there really the need to print both files if `diff` already shows the diff?

Also, I don't think we should print an error here. The caller probably prints a more relevant one anyway. And the name of the function does not even indicate that we're doing this to handle an error. It just sounds like a generic differ.

I think that the function should simply print the diff to stdout. The caller can easily redirect it to stderr with `diff_files ... >&2` if this is really an error.

Reply:
Same for `diff_values` above.

Reply:
Makes sense!